pkg/jwt/configuration.go

```diff
@@ -102,3 +120,52 @@ func NewSignatureValidatorFromJSONWebKeySet(jwks *jose.JSONWebKeySet) (Signature
 
 	return NewDemultiplexingSignatureValidator(namedSignatureValidators, allSignatureValidators), nil
 }
+
+// NewSignatureValidatorFromJSONWebKeySetFile creates a new
+// SignatureValidator capable of validating JWTs matching keys contained
+// in a JSON Web Key Set read from a file. The content of the file is
+// periodically refreshed.
+func NewSignatureValidatorFromJSONWebKeySetFile(path string, group program.Group) (SignatureValidator, error) {
+	internalValidator, err := getJwksFromFile(path)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Unable to read JWKS content from file at %s", path)
+	}
+	forwardingValidator := NewForwardingSignatureValidator(internalValidator)
+
+	group.Go(func(ctx context.Context, siblingsGroup, dependenciesGroup program.Group) error {
+		t := time.NewTicker(300 * time.Second)
+		defer t.Stop()
+
+		for {
+			select {
+			case <-t.C:
+				internalValidator, err := getJwksFromFile(path)
+				if err != nil {
+					log.Printf("Failed to read JWKS content from file at %s: %s", path, err)
```

Comment:
For the first `%s`, could you please use `%#v`? That way the string ends up getting quoted. Same the `status.Errorf()` above.